src/mlpack/prereqs.hpp

```diff
@@ -73,6 +73,23 @@ using enable_if_t = typename enable_if<B, T>::type;
 #endif
 #endif
 
+// Backport std::any from C+17 to C++11 to replace boost::any.
+// Use mnmlstc backport implementation only if compiler does not
+// support C++17.
+#if __cplusplus < 201703L
+  #include <mlpack/core/std_backport/any.hpp>
+  #include <mlpack/core/std_backport/string_view.hpp>
+  #define ANY core::v2::any
+  #define ANY_CAST core::v2::any_cast
+  #define STRING_VIEW core::v2::string_view
```

Comment:
Would it make sense to put them into the `std` namespace and just use `any_cast` etc. instead of our own alias?

Reply:
I am not sure what you mean, we can not use `std` since `string_view` and `any` are part of the C++17 standard library and not C++11.
I am using a backported version for these C++17 features, here is the link toward the backport:
https://github.com/mnmlstc/core
I have used aliases because on windows we are using directly C++ 17.

Reply:
Right now you put the functionality into `core::v2::` but you could also put it into `std::`, and use `any_cast` instead of `ANY_CAST`. Maybe there is a good reason to create an extra alias, that I missed here.

Reply:
I thought about this a bit too---personally I think it could be fine either way.  Ideally we would not be including any of the backported stuff if the compiler reports that it has C++17 or newer.  So, it could result in nicer code to just put it in `std::` directly; that will make transitioning away from the backports easier too.